### DISCUSSION TITLE
SetDefaultRenderController

### DISCUSSION THREAD

**Discussion by erikjanwestendorp** [umbraco/Umbraco-CMS#10608] [Features and ideas]
Today I had a conversation with the documentation team about custom controllers and the [PR ](https://github.com/umbraco/UmbracoDocs/pull/3265) to the documentation. In Umbraco 8 there was an extension method to change the default RenderMvcController. It seems it no longer exists. Would it be an idea to create a new Extension Method for v9 that would allow you to replace the default RenderController?

e.g.

```csharp
  public static IUmbracoBuilder SetDefaultRenderController(this IUmbracoBuilder builder)
  {
      var umbracoRenderingDefaults = builder.Services.FirstOrDefault(descriptor => descriptor.ServiceType == typeof(UmbracoRenderingDefaults));
      builder.Services.Remove(umbracoRenderingDefaults);

      builder.Services.AddSingleton<IUmbracoRenderingDefaults, MyRenderingDefaults>();

      return builder;
  }
```
See also the [article](https://our.umbraco.com/Documentation/Reference/Routing/custom-controllers-v9).

**Comment by Shazwazza** [comment]
It would be better to change UmbracoRenderingDefaults to be IOptions and remove IUmbracoRenderingDefaults. Then you can just configure options how you normally would and change the UmbracoRenderingDefaults.DefaultControllerType value.

**Reply by warrenbuckley** [reply]
PR for this is done here
https://github.com/umbraco/Umbraco-CMS/pull/10614
